### ISSUE TITLE
--warp-slot breaks solana-test-validator on Solana 1.14

### ISSUE THREAD

**Issue by ARitz-Cracker** [solana-labs/solana#30258]
Using the `--warp-slot` option on solana-test validator will cause the validator to not process any transactions.

Steps to reproduce:
1. Run `solana-test-validator`
2. Terminate it after it initializes
3. Run `solana-test-validator --warp-slot 20`

Currently running `1.14.15`

**Comment by bw-solana** [comment]
Tried the steps and I'm seeing transactions increment according to the CLI output:
```
sol@my-machine:~/solana$ solana-test-validator
Ledger location: test-ledger
Log: test-ledger/validator.log
⠄ Initializing...
⠠ Initializing...
Identity: 5FTijxLUfD6DZLV9j7VTi75DY1NuqnXU5H3tanmNFM7e
Genesis Hash: 4DBgH7YiJk1H64Ji9RNqHBrNVjYXNTcTRgdNcKVW28bK
Version: 1.10.30
Shred Version: 56389
Gossip Address: 127.0.0.1:1024
TPU Address: 127.0.0.1:1027
JSON RPC URL: http://127.0.0.1:8899
⠤ 00:00:02 | Processed Slot: 3 | Confirmed Slot: 3 | Finalized Slot: 0 | Full Snapshot Slot: - | Incremental Snapshot Slot: - | Transactions: 2 | ◎499.999990000
^C
sol@dev-equinix-chicago-4:~/solana$ solana-test-validator --warp-slot 20
--faucet-sol argument ignored, ledger already exists
Ledger location: test-ledger
Log: test-ledger/validator.log
⠲ Initializing...
⠄ Initializing...
Identity: 5FTijxLUfD6DZLV9j7VTi75DY1NuqnXU5H3tanmNFM7e
Genesis Hash: 4DBgH7YiJk1H64Ji9RNqHBrNVjYXNTcTRgdNcKVW28bK
Version: 1.10.30
Shred Version: 56389
Gossip Address: 127.0.0.1:1024
TPU Address: 127.0.0.1:1027
JSON RPC URL: http://127.0.0.1:8899
⠐ 00:00:35 | Processed Slot: 105 | Confirmed Slot: 105 | Finalized Slot: 73 | Full Snapshot Slot: 20 | Incremental Snapshot Slot: - | Transactions: 84 | ◎499.999580000
```

If you're slow killing the initial validator, then the second call will fail because we can only warp to a future slot (i.e. if the first instance makes it past slot 20, we cannot warp to slot 20 with the second instance).

Were you seeing something different?

**Comment by CriesofCarrots** [comment]
@bw-solana, from your output: `Version: 1.10.30`

This repros for me on v1.15.2:
```
$ solana-install init 1.15.2
  ✨ 1.15.2 initialized
$ solana-test-validator --reset --warp-slot 20
Ledger location: test-ledger
Log: test-ledger/validator.log
⠂ Initializing...
Bus error: 10
```

And on v1.14.16 it seems to simply hang at slot 20:
```
$ solana-install init 1.14.16
  ✨ 1.14.16 initialized
$ solana-test-validator --reset --warp-slot 20
Ledger location: test-ledger
Log: test-ledger/validator.log
⠐ Initializing...
⠋ Initializing...
⠁ Initializing...
⠚ Initializing...
⠒ Initializing...
⠄ Initializing...
⠠ Initializing...
⠖ Initializing...
⠒ Initializing...
⠈ Initializing...
Identity: 69BsCtTFeJDzgEB12YHDWGFMe9BezypotP1CQvE4mJzR
Genesis Hash: GSB8fKhzFf89M2TZaUKqVE49WUX1QJhkv13vKqzZ3MCi
Version: 1.14.16
Shred Version: 16939
Gossip Address: 127.0.0.1:1024
TPU Address: 127.0.0.1:1027
JSON RPC URL: http://127.0.0.1:8899
⠋ 00:00:29 | Processed Slot: 20 | Confirmed Slot: 20 | Finalized Slot: 20 | Full Snapshot Slot:
^C
```

**Comment by ARitz-Cracker** [comment]
> Tried the steps and I'm seeing transactions increment according to the CLI output:
> 
> ```
> sol@my-machine:~/solana$ solana-test-validator
> Ledger location: test-ledger
> Log: test-ledger/validator.log
> ⠄ Initializing...
> ⠠ Initializing...
> Identity: 5FTijxLUfD6DZLV9j7VTi75DY1NuqnXU5H3tanmNFM7e
> Genesis Hash: 4DBgH7YiJk1H64Ji9RNqHBrNVjYXNTcTRgdNcKVW28bK
> Version: 1.10.30
> Shred Version: 56389
> Gossip Address: 127.0.0.1:1024
> TPU Address: 127.0.0.1:1027
> JSON RPC URL: http://127.0.0.1:8899
> ⠤ 00:00:02 | Processed Slot: 3 | Confirmed Slot: 3 | Finalized Slot: 0 | Full Snapshot Slot: - | Incremental Snapshot Slot: - | Transactions: 2 | ◎499.999990000
> ^C
> sol@dev-equinix-chicago-4:~/solana$ solana-test-validator --warp-slot 20
> --faucet-sol argument ignored, ledger already exists
> Ledger location: test-ledger
> Log: test-ledger/validator.log
> ⠲ Initializing...
> ⠄ Initializing...
> Identity: 5FTijxLUfD6DZLV9j7VTi75DY1NuqnXU5H3tanmNFM7e
> Genesis Hash: 4DBgH7YiJk1H64Ji9RNqHBrNVjYXNTcTRgdNcKVW28bK
> Version: 1.10.30
> Shred Version: 56389
> Gossip Address: 127.0.0.1:1024
> TPU Address: 127.0.0.1:1027
> JSON RPC URL: http://127.0.0.1:8899
> ⠐ 00:00:35 | Processed Slot: 105 | Confirmed Slot: 105 | Finalized Slot: 73 | Full Snapshot Slot: 20 | Incremental Snapshot Slot: - | Transactions: 84 | ◎499.999580000
> ```
> 
> If you're slow killing the initial validator, then the second call will fail because we can only warp to a future slot (i.e. if the first instance makes it past slot 20, we cannot warp to slot 20 with the second instance).
> 
> Were you seeing something different?

Your version of Solana is outdated, 1.13 branch is the majority on Mainnet with 1.14 slowly taking over, 1.15 is being used on testnet. You are running 1.10.

**Comment by bw-solana** [comment]
Looks like this is caused by accounts hash/capitalization mismatch between Storage (0 lamports) and Index (lots of lamports) based data sources. v1.14 introduced a call to update accounts hash from `warp_from_parent`. It's not clear to me if we have mismatch in previous versions but just never noticed because we weren't checking.

Might need assistance from @brooksprumo or @jeffwashington to understand how to proceed.

**Comment by davidwei19930915** [comment]
Hello, are you interested in creating a new online virtual currency, which is rarer than traditional virtual coins, can drive manufacturing and employment problems for ordinary people, and more importantly, it is more suitable for Americans. And it has a lot to do with Starlink.  
If you are interested, please contact me My email address is <daivdwei19930915@outlook.com>

**Comment by bw-solana** [comment]
It appears the second (v1.14) issue is we are asserting in replay because we cannot find the root slot (0) in the ancestors list for current slot (20). If we just ignore this assert, we assume slot 20 is not locked out and everything seems to proceed just fine.

Need to figure out a way to either:
1. Get the root slot into the ancestor list for the slot we warp to
2. Ignore this assert in the case where we have warped slot forward

**Comment by bw-solana** [comment]
Closing as fix has been pushed and backported to v1.14 and v1.15 branches
